spec/acceptance/resolved_spec.rb

```diff
@@ -12,8 +12,11 @@
       }
       PUPPET
       apply_manifest(pp, catch_failures: true)
-      # RedHat 7, 9 and newer installs package first run before fact $facts['internal_services'] is set
-      apply_manifest(pp, catch_failures: true) if fact('os.release.major') != '8' && (fact('os.family') == 'RedHat')
+      # RedHat 7, 9, Debian 12 and newer installs package first run before fact $facts['internal_services'] is set
+      if (fact('os.release.major') != '8' && (fact('os.family') == 'RedHat')) ||
+         ((fact('os.name') == 'Debian' && fact('os.release.major').to_i >= 12))
+        apply_manifest(pp, catch_failures: true)
+      end
```

Comment:
For systemd-networkd I extracted this to a separate variable:
https://github.com/voxpupuli/puppet-systemd/blob/5f64bc177c6c1a39b557292840a46c7954dc9f05/spec/acceptance/networkd_spec.rb#L6

Then I reused it to check if the package is installed:
https://github.com/voxpupuli/puppet-systemd/blob/5f64bc177c6c1a39b557292840a46c7954dc9f05/spec/acceptance/networkd_spec.rb#L32